### ISSUE TITLE
Main branch after 1.0.0 releases

### ISSUE THREAD

**Issue by BeepBot99** [Mesure73L/mesure.x10.mx#79]
### Issue Contents

When developing for 1.1.0 after 1.0.0 releases, we can't push to the `main` branch because then people who are doing the challenges would see changes.
After it releases, instead, there should be a `production` branch. All edits are made to the `main` branch, but the website is on the `production` branch. The main branch is merged into the production branch only when 1.1.0 is released. That way, people will only see the changes when they should, and they won't be doing the challenges while it is being developed.

**Comment by BeepBot99** [comment]
I can do this.

**Comment by Mesure73L** [comment]
Can we discuss this first?

**Comment by Mesure73L** [comment]
> When developing for 1.1.0 after 1.0.0 releases, we can't push to the main branch because then people who are doing the challenges would see changes.

What do you mean by the people who are doing the challenges would see changes?

> After it releases, instead, there should be a production branch. All edits are made to the main branch, but the website is on the production branch. The main branch is merged into the production branch only when 1.1.0 is released. That way, people will only see the changes when they should, and they won't be doing the challenges while it is being developed.

I don't understand this.

**Comment by BeepBot99** [comment]
> [!IMPORTANT]
> First of all, note what milestone this is in. It is not until March.
# Explanation
The main branch is what is seen on the website.
If, in between March and August, someone is completing a challenge on Twelve of Code, and we are pushing developmental updates to the `main` branch, they will see it. Therefore, after 1.0.0 releases, we should have a `production` branch which is what will be pushed to the website instead of the `main` branch. When we release an update or bugfix, will will merge the `main` branch into the `production` branch, and then and only then wil the changes go onto the website.

**Comment by Mesure73L** [comment]
Why not the `production` branch for development (I can set up pages on it) and `main` for what is on the website?

**Comment by BeepBot99** [comment]
No, then it would be called `staging`. `production` means what is what the user sees, although the naming is confusing. But yes, you can create a branch called staging **after 1.0.0 releases.**

**Comment by BeepBot99** [comment]
> I can set up pages on it

What do you mean? What pages?

**Comment by Mesure73L** [comment]
> No, then it would be called `staging`. `production` means what is what the user sees, although the naming is confusing. But yes, you can create a branch called staging **after 1.0.0 releases.**

Let's just use a `staging` branch

> > I can set up pages on it
> 
> What do you mean? What pages?

Imagine not knowing what GitHub pages are

**Comment by BeepBot99** [comment]
Oh. It is confusing when you say things such as pages and hub, because that is not what they are called without context.
You do not need to set up pages, although you may if you want to.
The reason there is usually a `production` branch instead of a `staging` one is because when creating a new branch, by default, it forks from `main`. If `main` is the production branch, you would want to fork from `staging` instead, and you would have to change it manually **every single time**.

**Comment by Mesure73L** [comment]
Changed this to low priority, high priority should be for things that are urgent. We have a lot of time for this.

**Comment by BeepBot99** [comment]
No, by high priority, I meant high priority after 1.0.0 releases.

**Comment by Mesure73L** [comment]
…which is not high priority right now

**Comment by BeepBot99** [comment]
Completed.
